include/swift/Remote/MetadataReader.h

```diff
@@ -806,28 +806,35 @@ class MetadataReader {
     }
     case MetadataKind::Function: {
       auto Function = cast<TargetFunctionTypeMetadata<Runtime>>(Meta);
+      auto numParameters = Function->getNumArguments();
 
       std::vector<FunctionParam<BuiltType>> Parameters;
       StoredPointer ArgumentAddress = MetadataAddress +
         sizeof(TargetFunctionTypeMetadata<Runtime>);
-      for (StoredPointer i = 0; i < Function->getNumArguments(); ++i,
-           ArgumentAddress += sizeof(StoredPointer)) {
-        StoredPointer FlaggedArgumentAddress;
+      StoredPointer ParameterFlagsAddress =
+          ArgumentAddress + (numParameters * sizeof(StoredPointer));
+
+      for (StoredPointer i = 0; i < numParameters; ++i,
+                         ArgumentAddress += sizeof(StoredPointer),
+                         ParameterFlagsAddress += sizeof(StoredPointer)) {
```

Comment:
I believe you need to be adding sizeof(uint32_t) here.

But in general, none of this should be necessary, since it's not reading memory outside of the true bounds of the FunctionTypeMetadata.  You just need to update the Function case in readMetadata so that it computes the true bounds correctly, just like how it's done for Tuples.  If you do that, you should be able to just use the normal access functions on TargetFunctionTypeMetadata (as long as they don't assert anything you haven't already checked!).

Reply:
I think I understand now what you are talking about - it's done for parameters this way because we need to read a point to metadata itself from the argument position, but for flags we don't have to do that because they are values themselves. Thanks again!